src/parser/utils.rs

```diff
@@ -300,14 +310,22 @@ pub fn elaborate_field_path(
                 _ => None,
             };
 
+            let id_span = exp.pos.unwrap();
+            let acc_span = acc.pos.unwrap();
+            let pos = TermPos::Original(RawSpan::fuse(id_span, acc_span).unwrap());
+
```

Comment:
This code looks to be strictly similar to the one above. Can it be taken out of the pattern matching to avoid duplication?

Reply:
Hm. We need to get the `id` from the `FieldPathElem::Ident(..)`, and the `exp` from `FieldPathElem::Expr(..)` to get their positions. To do this before pattern matching , we would have to add a method to `FieldPathElem`, that returns the position or just do smaller pattern matching at the top to extract the position.

Reply:
Without a new method on `FieldPathElem`, we could at least take `acc_span` out of the pattern match.

Reply:
Ah, yes

Reply:
Ah, indeed, I missed that.

Reply:
Duplicating the pattern matching is still slightly less code in general, but in this case the duplicated part is so tiny that it's not very interesting indeed.